host/tests/rmfAudio_L3_TestCases/rmfAudio_test03_independentDataCheck.py

```diff
@@ -0,0 +1,138 @@
+
+        Args:
+            None
+        """
+        self.testName  = "test03_independentDataCheck"
+        super().__init__(self.testName, '3')
+
+    def testFunction(self):
+        """
+        The main test function that verifies primary and auxiliary data captures are independent.
+
+        This function:
+        - Starts Primary and Auxiliary Data captures.
+        - Tracks bytes received for both interfaces
+        - Writes the captured audio data to a wav file.
+        - Stops, starts in a mixed sequence to make sure the bytes received shows capture is independent
+
+        Returns:
+            bool: Final result of the test.
+        """
+        self.log.testStart(self.testName, '3')
+
+        # 1 for primary data capture (default), 2 for auxiliary data capture.
+        capture_type_primary = 1
+        capture_type_auxiliary = 2
+        sleep_duration = 2 #2 seconds to test
+        settings_update = 0 # settings_update=0 (no updates to default settings), capture_format=1, sampling_rate=1, fifo_size=1, threshold=1
+        # For settings update, set settings_update to 1, followed by value for capture_format, Sampling Frequency, FIFO size, threshold. 
+        # Use -1 for settings parameter that you want to retain default value for.
+        # Sample shows 1 for settings update, 0(racFormat_e16BitStereo) for capture format, 5(racFreq_e48000) for sampling rate, 16384 for FIFO and -1 to retain default threshold value.
+        # sample : self.testrmfAudio.updateSettings(capture_type, 1, 0, 5, 16384, -1)
+        test_type = 1 # test_type : 1 for byte counting tests (default), 2 for data tracking tests where audio data is captured.
+        aux_support = self.testrmfAudio.checkAuxiliarySupport()
+
+        if aux_support is True:
+            self.log.stepStart('Independent data test - check auxiliary runs independently')
+            self.testrmfAudio.openHandle(capture_type_primary)
+            # Retain default Settings in this sample
+            self.testrmfAudio.updateSettings(capture_type_primary, settings_update)
+            self.testrmfAudio.selectTestType(capture_type_primary, test_type)
+
+            self.testrmfAudio.openHandle(capture_type_auxiliary)
+            # Retain default Settings in this sample
+            self.testrmfAudio.updateSettings(capture_type_auxiliary, settings_update)
+            self.testrmfAudio.selectTestType(capture_type_auxiliary, test_type)
+
+            self.testrmfAudio.startCapture(capture_type_primary)
+            self.testrmfAudio.startCapture(capture_type_auxiliary)
+            
+            time.sleep(sleep_duration)
+            self.testrmfAudio.stopCapture(capture_type_primary)
+            #Primary stopped, auxiliary running. Verify if bytes received for auxiliary has increased.
+            primary_bytes_received_before, auxiliary_bytes_received_before = self.testrmfAudio.checkBytesReceived()

```

Comment:
primary_bytes_received_before where it is getting updated?

Reply:
Bytes received for primary and auxiliary interfaces is printed in C-test under menu "Check Bytes Received". 
checkBytesReceived in rmfAudioClass reads that value and extracts the bytes_received variables. 

Bytes received before and after for both interfaces are verified just below the highlighted lines here to validate that primary and aux interfaces run independently of each other.